From f2c236575f2e54407d7fff2a007d193a0223eabb Mon Sep 17 00:00:00 2001
From: Erwan Le Ray <erwan.leray@foss.st.com>
Date: Thu, 6 May 2021 16:11:43 +0200
Subject: [PATCH 106/529] dt-bindings: stm32: support new rs485 delays in
 nanoseconds

Add rs485-rts-delay-ns (in nanoseconds) property.
This extends the range of delay values for user to microseconds and
nanoseconds.

With fast data rates (1Mbaut or more), 1ms become quite long. Users need
to be able to set shorter delays than 1 ms in order not to slow down the
channel unnecessarily. This enables faster communication channels without
changing the baudrate.

Signed-off-by: Erwan Le Ray <erwan.leray@foss.st.com>
---
 Documentation/devicetree/bindings/serial/rs485.yaml         | 2 --
 Documentation/devicetree/bindings/serial/st,stm32-uart.yaml | 1 +
 2 files changed, 1 insertion(+), 2 deletions(-)

--- a/Documentation/devicetree/bindings/serial/rs485.yaml
+++ b/Documentation/devicetree/bindings/serial/rs485.yaml
@@ -31,8 +31,6 @@ properties:
 
   rs485-rts-delay-ns:
     description: prop-encoded-array <a b>
-    maxItems: 1
-    $ref: /schemas/types.yaml#/definitions/uint32-array
     items:
       items:
         - description: Delay between rts signal and beginning of data sent in
--- a/Documentation/devicetree/bindings/serial/st,stm32-uart.yaml
+++ b/Documentation/devicetree/bindings/serial/st,stm32-uart.yaml
@@ -61,6 +61,7 @@ properties:
   wakeup-source: true
 
   rs485-rts-delay: true
+  rs485-rts-delay-ns: true
   rs485-rts-active-low: true
   linux,rs485-enabled-at-boot-time: true
   rs485-rx-during-tx: true
